packages/react-component-library/src/components/DatePicker/utils/isDateDisabled.ts

```diff
@@ -5,11 +5,11 @@ import {
   isWithinInterval,
   startOfDay,
 } from 'date-fns'
-import { type Modifier } from 'react-day-picker'
+import { type Matcher } from 'react-day-picker'
 
 export const isDateDisabled = (
   date: Date,
-  disabledDays: Modifier | Modifier[]
+  disabledDays: Matcher | Matcher[] | undefined
```

Comment:
`daysOfWeek` is now `dayOfWeek` in the matchers so the function (and the tests) should be reflecting that...

However, I'm not sure this function is needed at all now – there is [useActiveModifiers](https://daypicker.dev/v8/api/functions/useActiveModifiers) and also [isMatch](https://daypicker.dev/v8/api/functions/isMatch) in react-day-picker v8. (Should hopefully be able to use `useActiveModifiers` to work out if `today` is disabled.)

Reply:
WIP adding test to supplement this.

Reply:
I've implemented a `isDateDisabled` function and backfilled tests accordingly.

The `DayPickerProps['disabled']` type actually supports a few other things beyond `Matcher[]`.